homeassistant/components/tts/amazon_polly.py

```diff
@@ -0,0 +1,160 @@
+    vol.Exclusive(CONF_PROFILE_NAME, ATTR_CREDENTIALS): cv.string,
+    # vol.Optional(CONF_VOICE, default=DEFAULT_VOICE):
+    #     vol.In(SUPPORTED_VOICES),
+    vol.Optional(CONF_OUTPUT_FORMAT, default=DEFAULT_OUTPUT_FORMAT):
+        vol.In(SUPPORTED_OUTPUT_FORMATS),
+    vol.Optional(CONF_SAMPLE_RATE): vol.All(cv.string,
+                                            vol.In(SUPPORTED_SAMPLE_RATES)),
+    vol.Optional(CONF_TEXT_TYPE, default=DEFAULT_TEXT_TYPE):
+        vol.In(SUPPORTED_TEXT_TYPES),
+})
+
+
+def get_engine(hass, config):
+    """Setup Amazon Polly speech component."""
+    # pylint: disable=import-error
+    output_format = config.get(CONF_OUTPUT_FORMAT)
+    sample_rate = config.get(CONF_SAMPLE_RATE,
+                             DEFAULT_SAMPLE_RATES[output_format])
+    if sample_rate not in SUPPORTED_SAMPLE_RATES_MAP.get(output_format):
+        _LOGGER.error("%s is not a valid sample rate for %s",
+                      sample_rate, output_format)
+        return None
+
+    import boto3
+
+    profile = config.get(CONF_PROFILE_NAME)
+
+    if profile is not None:
+        boto3.setup_default_session(profile_name=profile)
+
+    aws_config = {
+        CONF_REGION: config.get(CONF_REGION),
+        CONF_ACCESS_KEY_ID: config.get(CONF_ACCESS_KEY_ID),
+        CONF_SECRET_ACCESS_KEY: config.get(CONF_SECRET_ACCESS_KEY),
+    }
+
+    polly_client = boto3.client("polly", **aws_config)
+
+    all_voices = polly_client.describe_voices()
+
+    supported_languages = []
+
+    voice_language = "en-US"
+
+    for voice in all_voices.get("Voices"):
+        if voice.get("LanguageCode") not in supported_languages:
+            supported_languages.append(voice.get("LanguageCode"))
+        if voice.get("Id", "") == config.get(CONF_VOICE):
+            voice_language = "{}-{}".format(config.get(CONF_VOICE).lower(),
+                                            voice.get("LanguageCode"))
+
+    return AmazonPollyProvider(polly_client, config, supported_languages,
+                               voice_language)
+
+
+class AmazonPollyProvider(Provider):
+    """Amazon Polly speech api provider."""
+
+    def __init__(self, polly_client, config, supported_languages,
+                 voice_language):
+        """Initialize Amazon Polly provider for TTS."""
+        self.client = polly_client
+        self.config = config
+        self.all_languages = supported_languages
+        self.language = voice_language
+
+    @property
+    def supported_languages(self):
+        """List of supported languages."""
+        return self.all_languages
+
+    @property
+    def default_language(self):
+        """Default language."""
+        return self.language
+
+    @property
+    def supported_options(self):
+        """List of supported options."""
+        return [CONF_VOICE]
+
+    def get_tts_audio(self, message, language=None, options=None):
+        """Request TTS file from Polly."""
+        resp = self.client.synthesize_speech(
```

Comment:
Can you add a check of language is allow with given voice and abort if that constulation will not work (log error, return (None, None).